### ISSUE TITLE
Add alt text field to default Wagtail Image model and editing form

### ISSUE THREAD

**Issue by brylie** [wagtail/wagtail#6034]
Continued from #5291

The Wagtail images core `Image` model currently has an `alt` method that returns the image title as alt text. While it is possible to pass in alt text via the `image`  template tag, the alt text definition would more naturally live in the Image model, with the option to override it in the template definition.

![image](https://user-images.githubusercontent.com/17307/82079566-b9fbf680-96eb-11ea-80ef-a90cce7d62c2.png)

## Feature request
Add an optional `alt_text` field to the  Wagtail images Image model and editing form. Use the `alt_text` for the image tag `alt` text value when available. Also, allow users to override the `alt` text property by passing a value into the `{% image alt=value %}` tag.

## Alternative solutions
We ended up passing in the alt text value from a many-to-many table field. However, this is not ideal because the alt text must be re-defined any place the image is re-used.

## Additional context
With Wagtail's strong emphasis on accessibility, it seems at odds with the accessibility initiative that the `alt` text property is not part of the core Image model.

**Comment by gasman** [comment]
Hi @brylie,
I believe it's fairly common practice for people to add an alt_text field when defining a [custom image model](https://docs.wagtail.io/en/stable/advanced_topics/images/custom_image_model.html). However, Wagtail's default image model deliberately doesn't do this, for a fairly subtle reason - alt text should be [tailored to the context where the image is being used](http://www.ala.org/support/context-important-alt-text). You can see this principle in action in rich text fields, where editors are prompted for alt text at the point of inserting an image. Ideally, we'd want to encourage developers to follow the same pattern when building their own data models with images - so, for example, if they build an image gallery with a GalleryItem child model and InlinePanel, they ought to make the alt text a field on the GalleryItem model.

In practice, this is a big ask, both for developers and editors, and having a non-context-specific alt text field on the Image model is better than none at all. This leaves Wagtail in a tricky spot - if we were to offer the "fairly-good-practice" solution as standard (i.e. adding alt text as a field on Image), then this would get in the way of people implementing the "absolute-best-practice" approach (i.e. context-specific fields), as there would be two competing sources of alt text. Earlier this year I came up with some ideas and a proof-of-concept for how we might implement context-specific alt text, initially for StreamField image blocks, without placing a massive burden on developers or making the UI too clunky for editors - I'll try and get this written up as an RFC soon.

**Comment by brylie** [comment]
> I'll try and get this written up as an RFC soon.

Just pinging you about the image alt-text RFC. I'm working on a migration from Drupal to Wagtail and will be migrating a lot of images soon. :-)

**Comment by gasman** [comment]
@brylie RFC now created - https://github.com/wagtail/rfcs/pull/51
